### PR TITLE
Enable submitting a task to k8s using task_proc

### PR DESCRIPTION
**Pull request by nemacysts**
We aren't yet reading any of the events that task_proc is bubbling back
up to us, but that'll come next.

With this, we should be giving task_proc all the information it needs to
actually create a usable Pod.

### CODE REVIEW

**Comment by nemacysts** [comment]
we probably wanna grab https://github.com/Yelp/task_processing/pull/172 before merging this, but that won't change any of this code :)

File: tron/kubernetes.py [hunk 1]
```diff
@@ -292,23 +319,78 @@ def create_task(self, action_run_id: str, serializer, task_id: Optional[str] = N
...
+    def _check_connection(self) -> None:
+        """
+        Helper to ensure that the task_processing plugin is in a running state and event handling
+        is correctly setup in case we've disabled k8s at some point during operation.
+        """
+        if self.runner is None or self.runner.stopping:
+            log.info("k8s plugin never created or stopped, restarting.")
+            self.connect()
+        # re-add callbacks just in case they're missing
+        elif self.deferred is None or self.deferred.called:
```
Comment (jfongatyelp) [inline tron/kubernetes.py:360]: should this be an elif or just another if? Otherwise if we did have to reconnect above, we'd never re-check callbacks here

**Review by jfongatyelp — APPROVED** [review]
I still am not particularly familiar with how tron tasks are submitted to the cluster and how the pieces come together with taskproc yet, but I see these are pretty isolated codepaths for mesos vs kubernetes, so should be pretty safe, and those that are shared (eg actionrun) seem to be safe changes.

File: tron/core/actionrun.py [hunk 2]
```diff
@@ -193,6 +195,8 @@ class ActionRun(Observable):
     EXIT_STOP_KILL = -3
     EXIT_TRIGGER_TIMEOUT = -4
     EXIT_MESOS_DISABLED = -5
+    EXIT_KUBERNETES_DISABLED = -6
```
[the remark this answers is not in the document]
Comment (nemacysts) [reply tron/core/actionrun.py:198]: oh, good catch!

File: tron/kubernetes.py (hunk 1 above)
Comment (nemacysts) [reply tron/kubernetes.py:360]: if we did have to reconnect, that path will eventually also call `self.handle_next_event` and do this same check before calling `self.handle_next_event`

File: tron/core/actionrun.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nemacysts) [reply tron/core/actionrun.py:198]: added!

**Review by jfongatyelp — APPROVED** [review]

**Pull request merged**